### PR TITLE
Only handle indentation for open curly brace instead of all metacode

### LINKED ISSUE #1461: Hitting <Enter> after "@page" in a Razor page indents unnecessarily
image

The screenshot shows the editor immediately after `<Enter>`. `<Backspace>` once puts the cursor back at the beginning of the line.

### PR DESCRIPTION
**Pull request by ajaybhargavb**
Fixes #1461 

- Renamed a file
- Added a few tests for `GetDesiredIndentation`. Should probably add more tests later. Worth filing an issue?

Recommend reviewing with ?w=1

### CODE REVIEW

File: src/Microsoft.CodeAnalysis.Razor.Workspaces/DefaultRazorIndentationFactsService.cs [hunk 1]
```diff
@@ -39,52 +39,53 @@ internal class DefaultRazorIndentationFactsService : RazorIndentationFactsServic
 
             var simulatedChange = new SourceChange(previousLineEndIndex, 0, string.Empty);
             var owningSpan = LocateOwner(syntaxTree.Root, simulatedChange);
+            if (owningSpan.Kind == SpanKindInternal.Code || owningSpan.Kind == SpanKindInternal.None)
+            {
+                return null;
```
Comment (NTaylorMullen) [inline src/Microsoft.CodeAnalysis.Razor.Workspaces/DefaultRazorIndentationFactsService.cs]: Leave a comment about cases which result in this.

**Comment by rynowak** [comment]
> Added a few tests for GetDesiredIndentation. Should probably add more tests later. Worth filing an issue? 

Yeah please do

File: src/Microsoft.CodeAnalysis.Razor.Workspaces/DefaultRazorIndentationFactsService.cs [hunk 2]
```diff
@@ -39,52 +39,53 @@ internal class DefaultRazorIndentationFactsService : RazorIndentationFactsServic
...
                 {
-                    Block owningParent = owningChild.Parent;
-                    List<SyntaxTreeNode> children = new List<SyntaxTreeNode>(owningParent.Children);
-                    for (int i = 0; i < children.Count; i++)
+                    SyntaxTreeNode curChild = children[i];
+                    if (!curChild.IsBlock)
                     {
-                        SyntaxTreeNode curChild = children[i];
-                        if (!curChild.IsBlock)
+                        Span curSpan = curChild as Span;
```
Comment (rynowak) [inline src/Microsoft.CodeAnalysis.Razor.Workspaces/DefaultRazorIndentationFactsService.cs]: Can we rename this to something reasonable?

**Comment by NTaylorMullen** [comment]
Talked a bit more offline and there's a more general purpose fix for this which is making our indentation logic introspect on curly braces (not just meta-code). That was the initial intentions anyhow.

**Comment by ajaybhargavb** [comment]
🆙 📅

**Comment by NTaylorMullen** [comment]
Needs tests that validate the curly brace requirements.

**Comment by ajaybhargavb** [comment]
> Needs tests that validate the curly brace requirements.

@NTaylorMullen did you mean we need more tests as part of this PR rather than https://github.com/aspnet/Razor/issues/1698?

**Comment by NTaylorMullen** [comment]
@ajaybhargavb yes. That issue is more about adding more over-arching tests. This PR should have tests that revolve around what you added.

**Comment by ajaybhargavb** [comment]
🆙 📅

**Pull request merged**